pandas/tests/series/test_constructors.py

```diff
@@ -783,6 +783,18 @@ def test_constructor_dict(self):
         expected.iloc[1] = 1
         assert_series_equal(result, expected)
 
+    def test_constructor_dict_order(self):
```

Comment:
Maybe could decorate these tests with `@pytest.mark.skipif(not PY36, reason="Python 3.6 specific behavior")`.  Not sure if we're gaining much testing the non-PY36 path.  Would wait to see what others think before making this change though.

Reply:
Pandas has clearly defined ordering for python < 3.6 also (sort  by values). So it's nneeded to test this distinction and not use a skipif, IMO.

Reply:
@topper-123 let's go ahead and make 2 different tests here, 1 for >= 3.6 and one for less, they should skip on the alternative, same for the tests in frame.